### PR TITLE
Modernize classes of sandbox components

### PR DESCRIPTION
**Pull request by mawiesne**
- modernizes classes to use a Java8+ code style (generics, ...)
- rewrites some old c-style array declarations
- adds final modifier where applicable
- switches to enhanced for-loop pattern where applicable
- removes unnecessary / useless variable initializations
- fixes compareTo implementations that violated the contract for equality
- adds static modifier for inner classes
- replaces `String#indexOf(..) < ..` checks with `String#contains(..)`
- replaces StringBuffer with StringBuilder
- removes unused imports
- removes superfluous toString() calls
- modernizes resource handling along the path

### CODE REVIEW

File: opennlp-coref/src/main/java/opennlp/tools/coref/mention/DefaultParse.java [hunk 1]
```diff
@@ -36,12 +36,12 @@
...
-  public static String[] NAME_TYPES = {"person", "organization", "location", "date",
+  public static final String[] NAME_TYPES = {"person", "organization", "location", "date",
       "time", "percentage", "money"};
   
-  private Parse parse;
-  private int sentenceNumber;
-  private static Set<String> entitySet = new HashSet<String>(Arrays.asList(NAME_TYPES));
+  private final Parse parse;
+  private final int sentenceNumber;
+  private static final Set<String> entitySet = new HashSet<>(Arrays.asList(NAME_TYPES));
```
Comment (rzo1) [inline opennlp-coref/src/main/java/opennlp/tools/coref/mention/DefaultParse.java:44]: Do we want to adhere to naming conventions regarding capslock for constants?

File: opennlp-similarity/src/main/java/opennlp/tools/apps/relevanceVocabs/SynonymMap.java [hunk 2]
```diff
@@ -240,7 +240,7 @@ private HashMap<String,String[]> createIndex(Map<String,ArrayList<Integer>> word
        ArrayList<String> words = group2Words.get(group.get(i));
        for (int j=words.size(); --j >= 0; ) { // add all words
          String synonym = words.get(j); // note that w and word are interned
-         if (synonym != word) { // a word implicitly is its own synonym
+         if (!synonym.equals(word)) { // a word implicitly is its own synonym
```
Comment (rzo1) [inline opennlp-similarity/src/main/java/opennlp/tools/apps/relevanceVocabs/SynonymMap.java:243]: good catch

File: opennlp-similarity/src/main/java/opennlp/tools/doc_classifier/ClassifierTrainingSetIndexer.java [hunk 3]
```diff
@@ -31,18 +32,18 @@
...
-import org.apache.lucene.util.Version;
 import org.apache.tika.Tika;
 
 public class ClassifierTrainingSetIndexer {
   
-  public static String resourceDir = new File(".").getAbsolutePath().replace("/.", "") + "/src/main/resources";
-  public static String INDEX_PATH = "/classif", CLASSIF_TRAINING_CORPUS_PATH = "/training_corpus";
-  protected ArrayList<File> queue = new ArrayList<>();
-  Tika tika = new Tika();
+  public static final String resourceDir = new File(".").getAbsolutePath().replace("/.", "") + "/src/main/resources";
```
Comment (rzo1) [inline opennlp-similarity/src/main/java/opennlp/tools/doc_classifier/ClassifierTrainingSetIndexer.java:39]: `RESOURCE_DIR` ?

File: opennlp-similarity/src/main/java/opennlp/tools/jsmlearning/JSMLearnerOnLatticeWithAbduction.java [hunk 4]
```diff
@@ -17,26 +17,8 @@
 
 package opennlp.tools.jsmlearning;
```
Comment (rzo1) [inline opennlp-similarity/src/main/java/opennlp/tools/jsmlearning/JSMLearnerOnLatticeWithAbduction.java:18]: Looks like we need to enforce formatting rules as a next step.

**Review by rzo1 — APPROVED** [review]

File: opennlp-coref/src/main/java/opennlp/tools/coref/mention/DefaultParse.java (hunk 1 above)
Comment (mawiesne) [reply opennlp-coref/src/main/java/opennlp/tools/coref/mention/DefaultParse.java:44]: Sure, next iteration, next topic.

File: opennlp-similarity/src/main/java/opennlp/tools/doc_classifier/ClassifierTrainingSetIndexer.java (hunk 3 above)
Comment (mawiesne) [reply opennlp-similarity/src/main/java/opennlp/tools/doc_classifier/ClassifierTrainingSetIndexer.java:39]: See comment above, next round. :)

File: opennlp-similarity/src/main/java/opennlp/tools/jsmlearning/JSMLearnerOnLatticeWithAbduction.java (hunk 4 above)
Comment (mawiesne) [reply opennlp-similarity/src/main/java/opennlp/tools/jsmlearning/JSMLearnerOnLatticeWithAbduction.java:18]: Definitely...

**Review by kinow — APPROVED** [review]
Thank you!!! :1st_place_medal:

**Pull request merged**